In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

sns.set_theme(style='whitegrid')
plt.rcParams['figure.figsize'] = (10, 5)

#Carga del dataset desde archivo local
df = pd.read_csv('penguins_raw.csv')
                

# Diagnóstico inicial

Dimensiones   
Tipos de datos   
Nulos por columna   
Duplicados

In [ ]:
df.head ()

,studyName,Sample Number,Species,Region,Island,Stage,Individual ID,Clutch Completion,Date Egg,Culmen Length (mm),Culmen Depth (mm),Flipper Length (mm),Body Mass (g),Sex,Delta 15 N (o/oo),Delta 13 C (o/oo),Comments
0,PAL0708,1,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N1A1,Yes,2007-11-11,39.1,18.7,181.0,3750.0,MALE,NaN,NaN,Not enough blood for isotopes.
1,PAL0708,2,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N1A2,Yes,2007-11-11,39.5,17.4,186.0,3800.0,FEMALE,8.94956,-24.69454,NaN
2,PAL0708,3,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N2A1,Yes,2007-11-16,40.3,18.0,195.0,3250.0,FEMALE,8.36821,-25.33302,NaN
3,PAL0708,4,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N2A2,Yes,2007-11-16,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Adult not sampled.
4,PAL0708,5,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N3A1,Yes,2007-11-16,36.7,19.3,193.0,3450.0,FEMALE,8.76651,-25.32426,NaN


In [ ]:
df.info()  # Tipos, nulos, memoria

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 17 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   studyName            344 non-null    object 
 1   Sample Number        344 non-null    int64  
 2   Species              344 non-null    object 
 3   Region               344 non-null    object 
 4   Island               344 non-null    object 
 5   Stage                344 non-null    object 
 6   Individual ID        344 non-null    object 
 7   Clutch Completion    344 non-null    object 
 8   Date Egg             344 non-null    object 
 9   Culmen Length (mm)   342 non-null    float64
 10  Culmen Depth (mm)    342 non-null    float64
 11  Flipper Length (mm)  342 non-null    float64
 12  Body Mass (g)        342 non-null    float64
 13  Sex                  333 non-null    object 
 14  Delta 15 N (o/oo)    330 non-null    float64
 15  Delta 13 C (o/oo)    331 non-null    flo

In [ ]:
print(f"Filas: {df.shape[0]}, Columnas: {df.shape[1]}")

Filas: 344, Columnas: 17


In [ ]:
df.describe()

,Sample Number,Culmen Length (mm),Culmen Depth (mm),Flipper Length (mm),Body Mass (g),Delta 15 N (o/oo),Delta 13 C (o/oo)
count,344.000000,342.000000,342.000000,342.000000,342.000000,330.000000,331.000000
mean,63.151163,43.921930,17.151170,200.915205,4201.754386,8.733382,-25.686292
std,40.430199,5.459584,1.974793,14.061714,801.954536,0.551770,0.793961
min,1.000000,32.100000,13.100000,172.000000,2700.000000,7.632200,-27.018540
25%,29.000000,39.225000,15.600000,190.000000,3550.000000,8.299890,-26.320305
50%,58.000000,44.450000,17.300000,197.000000,4050.000000,8.652405,-25.833520
75%,95.250000,48.500000,18.700000,213.000000,4750.000000,9.172123,-25.062050
max,152.000000,59.600000,21.500000,231.000000,6300.000000,10.025440,-23.787670


In [ ]:
df.nunique()

studyName                3
Sample Number          152
Species                  3
Region                   1
Island                   3
Stage                    1
Individual ID          190
Clutch Completion        2
Date Egg                50
Culmen Length (mm)     164
Culmen Depth (mm)       80
Flipper Length (mm)     55
Body Mass (g)           94
Sex                      2
Delta 15 N (o/oo)      330
Delta 13 C (o/oo)      331
Comments                10
dtype: int64

In [ ]:
df.isnull().sum().sort_values(ascending=False)

Comments               290
Delta 15 N (o/oo)       14
Delta 13 C (o/oo)       13
Sex                     11
Culmen Length (mm)       2
Flipper Length (mm)      2
Culmen Depth (mm)        2
Body Mass (g)            2
studyName                0
Date Egg                 0
Clutch Completion        0
Individual ID            0
Stage                    0
Island                   0
Region                   0
Species                  0
Sample Number            0
dtype: int64

In [ ]:
df.duplicated().sum()

np.int64(0)

In [ ]:
df.dtypes

studyName               object
Sample Number            int64
Species                 object
Region                  object
Island                  object
Stage                   object
Individual ID           object
Clutch Completion       object
Date Egg                object
Culmen Length (mm)     float64
Culmen Depth (mm)      float64
Flipper Length (mm)    float64
Body Mass (g)          float64
Sex                     object
Delta 15 N (o/oo)      float64
Delta 13 C (o/oo)      float64
Comments                object
dtype: object

In [ ]:
for col in df.columns:
    display(f"{col}: {df[col].nunique()} valores únicos")

'studyName: 3 valores únicos'

'Sample Number: 152 valores únicos'

'Species: 3 valores únicos'

'Region: 1 valores únicos'

'Island: 3 valores únicos'

'Stage: 1 valores únicos'

'Individual ID: 190 valores únicos'

'Clutch Completion: 2 valores únicos'

'Date Egg: 50 valores únicos'

'Culmen Length (mm): 164 valores únicos'

'Culmen Depth (mm): 80 valores únicos'

'Flipper Length (mm): 55 valores únicos'

'Body Mass (g): 94 valores únicos'

'Sex: 2 valores únicos'

'Delta 15 N (o/oo): 330 valores únicos'

'Delta 13 C (o/oo): 331 valores únicos'

'Comments: 10 valores únicos'


1. Que muestra este diagnostico general.
   
Tras la observación inicial, detectamos que:
    1.1 Hay pinguinos que tienen el mismo ID pero que viven en islas diferentes, tienen un sexo distinto, especies diferentes y solo observamos que aparentemente corresponden a distintos años de cría. Suponemos que es porque se reciclan los IDs cada año.
    Observamos que al ordenar por fecha, van todos los pinguinos por pareja. Teniendo en cuenta este patrón, podemos limpiar los valores nulos en algunas columnas.
    1.2 Hay valores nulos principalmente en: 
Comments               290   
Delta 15 N (o/oo)       14   
Delta 13 C (o/oo)       13   
Sex                     11   
Culmen Length (mm)       2   
Flipper Length (mm)      2   
Culmen Depth (mm)        2   
Body Mass (g)            2   
    1.3 Aparece el dtype "object" recurrentemente, aunque consideramos que no nos afectará en la estadística.
    1.4 Los datos Delta 15 N, Delta 13 C se corresponden a qué comen y dónde comen los pinguinos, esto nos puede servir para categorizarlos. Puede ser un indicador de las características de la fauna de cada territorio.
    1.5 Observamos que en el dataset aparecen las características morfológicas y que estas influyen en dónde y cómo pescan y las características de su dieta.
    
2. Que problema de calidad os parece mas relevante.

A la hora de obtener resultados del análisis, el hecho de que los IDs sean compartidos genera confusión y dificultad de interpretación. 


3. Por que ese problema le importa al cliente.

Es un riesgo de calidad del dato y puede implicar sesgos y errores de muestreo. Además, es una recomendación concreta para mejorar futuras campañas.

In [ ]:
#Ejemplo de individuos distintos con mismo ID.

df[df["Individual ID"] == "N13A1"]

,studyName,Sample Number,Species,Region,Island,Stage,Individual ID,Clutch Completion,Date Egg,Culmen Length (mm),Culmen Depth (mm),Flipper Length (mm),Body Mass (g),Sex,Delta 15 N (o/oo),Delta 13 C (o/oo),Comments
24,PAL0708,25,Adelie Penguin (Pygoscelis adeliae),Anvers,Biscoe,"Adult, 1 Egg Stage",N13A1,Yes,2007-11-10,38.8,17.2,180.0,3800.0,MALE,9.63954,-25.29856,NaN
202,PAL0809,51,Gentoo penguin (Pygoscelis papua),Anvers,Biscoe,"Adult, 1 Egg Stage",N13A1,Yes,2008-11-04,46.6,14.2,210.0,4850.0,FEMALE,8.38289,-26.86352,NaN
238,PAL0910,87,Gentoo penguin (Pygoscelis papua),Anvers,Biscoe,"Adult, 1 Egg Stage",N13A1,Yes,2009-11-20,43.4,14.4,218.0,4600.0,FEMALE,8.23468,-26.18599,NaN


# Limpieza

Datos que necesitamos:   

    Especies: Cómo se distribuyen (Island, region)   
              Diferencias morfológicas relevantes (Sex, Culmen length, Culmen depth, flipper length, body mass)   
    Perfiles biológicos: (Species, Culmen length, Culmen depth, flipper length, body mass)   
    

### Tabla de decisiones de calidad del dato

Completad esta tabla durante la limpieza y actualizadla si cambian vuestras decisiones.

| Problema detectado | Columna afectada | Decision tomada | Justificacion | Impacto esperado |
|---|---|---|---|---|
| 2 Pingüinos con estádisticas nulas | Culmen Length (mm)    Culmen Depth (mm)    Flipper Length (mm)    Body Mass (g)| Eliminación de datos | No son relevantes | Ninguno |
| Pinguïnos con sexo = NaN| Sex | Reemplazar NaN por "indefinido" | Poder trabajar sin valores NaN | Estádisticas más fiables|
| Columnas con información no relevante| Region Clutch Completion Comments| Eliminar columnas| Datos no relevantes| Dataset más visual|


In [ ]:
#Observación de nulos

df[df["Culmen Length (mm)"].isnull()]



,studyName,Sample Number,Species,Region,Island,Stage,Individual ID,Clutch Completion,Date Egg,Culmen Length (mm),Culmen Depth (mm),Flipper Length (mm),Body Mass (g),Sex,Delta 15 N (o/oo),Delta 13 C (o/oo),Comments
3,PAL0708,4,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N2A2,Yes,2007-11-16,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Adult not sampled.
271,PAL0910,120,Gentoo penguin (Pygoscelis papua),Anvers,Biscoe,"Adult, 1 Egg Stage",N38A2,No,2009-12-01,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Adult not sampled. Nest never observed with fu...


In [ ]:
#Observación de nulos

df[df["Delta 15 N (o/oo)"].isnull()]


,studyName,Sample Number,Species,Region,Island,Stage,Individual ID,Clutch Completion,Date Egg,Culmen Length (mm),Culmen Depth (mm),Flipper Length (mm),Body Mass (g),Sex,Delta 15 N (o/oo),Delta 13 C (o/oo),Comments
0,PAL0708,1,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N1A1,Yes,2007-11-11,39.1,18.7,181.0,3750.0,MALE,NaN,NaN,Not enough blood for isotopes.
3,PAL0708,4,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N2A2,Yes,2007-11-16,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Adult not sampled.
8,PAL0708,9,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N5A1,Yes,2007-11-09,34.1,18.1,193.0,3475.0,NaN,NaN,NaN,No blood sample obtained.
11,PAL0708,12,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N6A2,Yes,2007-11-09,37.8,17.3,180.0,3700.0,NaN,NaN,NaN,No blood sample obtained.
12,PAL0708,13,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N7A1,Yes,2007-11-15,41.1,17.6,182.0,3200.0,FEMALE,NaN,NaN,Not enough blood for isotopes.
13,PAL0708,14,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N7A2,Yes,2007-11-15,38.6,21.2,191.0,3800.0,MALE,NaN,NaN,Not enough blood for isotopes.
15,PAL0708,16,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N8A2,Yes,2007-11-16,36.6,17.8,185.0,3700.0,FEMALE,NaN,NaN,Not enough blood for isotopes.
39,PAL0708,40,Adelie Penguin (Pygoscelis adeliae),Anvers,Dream,"Adult, 1 Egg Stage",N25A2,No,2007-11-13,39.8,19.1,184.0,4650.0,MALE,NaN,NaN,Nest never observed with full clutch. Not enou...
41,PAL0708,42,Adelie Penguin (Pygoscelis adeliae),Anvers,Dream,"Adult, 1 Egg Stage",N26A2,Yes,2007-11-16,40.8,18.4,195.0,3900.0,MALE,NaN,NaN,Not enough blood for isotopes.
46,PAL0708,47,Adelie Penguin (Pygoscelis adeliae),Anvers,Dream,"Adult, 1 Egg Stage",N29A1,Yes,2007-11-13,41.1,19.0,182.0,3425.0,MALE,NaN,NaN,Not enough blood for isotopes.


In [ ]:
#Observación de nulos

df[df["Delta 13 C (o/oo)"].isnull()]

,studyName,Sample Number,Species,Region,Island,Stage,Individual ID,Clutch Completion,Date Egg,Culmen Length (mm),Culmen Depth (mm),Flipper Length (mm),Body Mass (g),Sex,Delta 15 N (o/oo),Delta 13 C (o/oo),Comments
0,PAL0708,1,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N1A1,Yes,2007-11-11,39.1,18.7,181.0,3750.0,MALE,NaN,NaN,Not enough blood for isotopes.
3,PAL0708,4,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N2A2,Yes,2007-11-16,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Adult not sampled.
8,PAL0708,9,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N5A1,Yes,2007-11-09,34.1,18.1,193.0,3475.0,NaN,NaN,NaN,No blood sample obtained.
11,PAL0708,12,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N6A2,Yes,2007-11-09,37.8,17.3,180.0,3700.0,NaN,NaN,NaN,No blood sample obtained.
12,PAL0708,13,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N7A1,Yes,2007-11-15,41.1,17.6,182.0,3200.0,FEMALE,NaN,NaN,Not enough blood for isotopes.
13,PAL0708,14,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N7A2,Yes,2007-11-15,38.6,21.2,191.0,3800.0,MALE,NaN,NaN,Not enough blood for isotopes.
15,PAL0708,16,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N8A2,Yes,2007-11-16,36.6,17.8,185.0,3700.0,FEMALE,NaN,NaN,Not enough blood for isotopes.
39,PAL0708,40,Adelie Penguin (Pygoscelis adeliae),Anvers,Dream,"Adult, 1 Egg Stage",N25A2,No,2007-11-13,39.8,19.1,184.0,4650.0,MALE,NaN,NaN,Nest never observed with full clutch. Not enou...
41,PAL0708,42,Adelie Penguin (Pygoscelis adeliae),Anvers,Dream,"Adult, 1 Egg Stage",N26A2,Yes,2007-11-16,40.8,18.4,195.0,3900.0,MALE,NaN,NaN,Not enough blood for isotopes.
46,PAL0708,47,Adelie Penguin (Pygoscelis adeliae),Anvers,Dream,"Adult, 1 Egg Stage",N29A1,Yes,2007-11-13,41.1,19.0,182.0,3425.0,MALE,NaN,NaN,Not enough blood for isotopes.


In [2]:
#Eliminar filas

df_limpio = df.dropna(subset=["Culmen Length (mm)"])

In [3]:
#Eliminar filas

df_limpio = df.dropna(subset=["Delta 15 N (o/oo)"])

In [ ]:
print('Filas originales:', len(df))
print('Filas tras limpieza:', len(df_limpio))
print('\nNulos tras limpieza:')
print(df_limpio.isna().sum())

Filas originales: 344
Filas tras limpieza: 330

Nulos tras limpieza:
studyName                0
Sample Number            0
Species                  0
Region                   0
Island                   0
Stage                    0
Individual ID            0
Clutch Completion        0
Date Egg                 0
Culmen Length (mm)       0
Culmen Depth (mm)        0
Flipper Length (mm)      0
Body Mass (g)            0
Sex                      6
Delta 15 N (o/oo)        0
Delta 13 C (o/oo)        0
Comments               290
dtype: int64


In [ ]:
countsex = df['Sex'].value_counts()

print(countsex)
# Acceso específico:
print(f"Male: {countsex.get('macho', 0)}")
print(f"Female: {countsex.get('hembra', 0)}")

Sex
MALE      168
FEMALE    165
Name: count, dtype: int64
Male: 0
Female: 0


In [ ]:
#Otra forma de verlo

male = (df["Sex"] == "MALE").sum()

female = (df["Sex"] == "FEMALE").sum()

print(f"Male:  {male}")
print(f"Female:  {female}")

Male:  168
Female:  165


In [6]:
#Rellenamos los nulos de Sex con "indefinido"


df_limpio['Sex'] = df_limpio['Sex'].fillna('indefinido')

df_limpio

/var/folders/bc/xktj9y7x4hgdxrzyj55hz_th0000gn/T/ipykernel_43468/400883569.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_limpio['Sex'] = df_limpio['Sex'].fillna('indefinido')


,studyName,Sample Number,Species,Island,Stage,Individual ID,Date Egg,Culmen Length (mm),Culmen Depth (mm),Flipper Length (mm),Body Mass (g),Sex,Delta 15 N (o/oo),Delta 13 C (o/oo)
1,PAL0708,2,Adelie Penguin (Pygoscelis adeliae),Torgersen,"Adult, 1 Egg Stage",N1A2,2007-11-11,39.5,17.4,186.0,3800.0,FEMALE,8.94956,-24.69454
2,PAL0708,3,Adelie Penguin (Pygoscelis adeliae),Torgersen,"Adult, 1 Egg Stage",N2A1,2007-11-16,40.3,18.0,195.0,3250.0,FEMALE,8.36821,-25.33302
4,PAL0708,5,Adelie Penguin (Pygoscelis adeliae),Torgersen,"Adult, 1 Egg Stage",N3A1,2007-11-16,36.7,19.3,193.0,3450.0,FEMALE,8.76651,-25.32426
5,PAL0708,6,Adelie Penguin (Pygoscelis adeliae),Torgersen,"Adult, 1 Egg Stage",N3A2,2007-11-16,39.3,20.6,190.0,3650.0,MALE,8.66496,-25.29805
6,PAL0708,7,Adelie Penguin (Pygoscelis adeliae),Torgersen,"Adult, 1 Egg Stage",N4A1,2007-11-15,38.9,17.8,181.0,3625.0,FEMALE,9.18718,-25.21799
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
339,PAL0910,64,Chinstrap penguin (Pygoscelis antarctica),Dream,"Adult, 1 Egg Stage",N98A2,2009-11-19,55.8,19.8,207.0,4000.0,MALE,9.70465,-24.53494
340,PAL0910,65,Chinstrap penguin (Pygoscelis antarctica),Dream,"Adult, 1 Egg Stage",N99A1,2009-11-21,43.5,18.1,202.0,3400.0,FEMALE,9.37608,-24.40753
341,PAL0910,66,Chinstrap penguin (Pygoscelis antarctica),Dream,"Adult, 1 Egg Stage",N99A2,2009-11-21,49.6,18.2,193.0,3775.0,MALE,9.46180,-24.70615
342,PAL0910,67,Chinstrap penguin (Pygoscelis antarctica),Dream,"Adult, 1 Egg Stage",N100A1,2009-11-21,50.8,19.0,210.0,4100.0,MALE,9.98044,-24.68741


In [ ]:
#Comprobación

df_limpio[df_limpio["Individual ID"] == "N5A2"]

,studyName,Sample Number,Species,Region,Island,Stage,Individual ID,Clutch Completion,Date Egg,Culmen Length (mm),Culmen Depth (mm),Flipper Length (mm),Body Mass (g),Sex,Delta 15 N (o/oo),Delta 13 C (o/oo),Comments
9,PAL0708,10,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N5A2,Yes,2007-11-09,42.0,20.2,190.0,4250.0,indefinido,9.13362,-25.09368,No blood sample obtained for sexing.
191,PAL0809,40,Gentoo penguin (Pygoscelis papua),Anvers,Biscoe,"Adult, 1 Egg Stage",N5A2,Yes,2008-11-09,48.7,15.7,208.0,5350.0,MALE,8.14776,-26.84506,NaN


In [4]:
df_limpio.drop(columns=["Region"], inplace=True)
df_limpio.drop(columns=["Clutch Completion"], inplace=True)
df_limpio.drop(columns=["Comments"], inplace=True)

/var/folders/bc/xktj9y7x4hgdxrzyj55hz_th0000gn/T/ipykernel_43468/1216441014.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_limpio.drop(columns=["Region"], inplace=True)
/var/folders/bc/xktj9y7x4hgdxrzyj55hz_th0000gn/T/ipykernel_43468/1216441014.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_limpio.drop(columns=["Clutch Completion"], inplace=True)
/var/folders/bc/xktj9y7x4hgdxrzyj55hz_th0000gn/T/ipykernel_43468/1216441014.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user

In [7]:
#Comprobación de nulos actualizados

print('\nNulos tras limpieza:')
print(df_limpio.isna().sum())


Nulos tras limpieza:
studyName              0
Sample Number          0
Species                0
Island                 0
Stage                  0
Individual ID          0
Date Egg               0
Culmen Length (mm)     0
Culmen Depth (mm)      0
Flipper Length (mm)    0
Body Mass (g)          0
Sex                    0
Delta 15 N (o/oo)      0
Delta 13 C (o/oo)      0
dtype: int64


Tras realizar la limpieza de nulos, generamos un nuevo csv optimizado con los datos limpios para trabajar con este documento a partir de ahora y evitar errores.

In [ ]:
df_limpio.to_csv("penguins_limpio.csv")